tracer/src/Datadog.Trace/DuckTyping/DuckType.Statics.cs

```diff
@@ -185,6 +187,74 @@ static ModuleBuilder CreateModuleBuilder(string name, Assembly targetAssembly)
             }
         }
 
+        private static Type? GetTypeFromPartialName(string partialName, bool throwOnError = false)
+        {
+            Type? type = null;
+            ExceptionDispatchInfo? loadException = null;
+
+            try
+            {
+                // Let's try to find the type using the partial name first.
+                type = Type.GetType(partialName, throwOnError: throwOnError);
+            }
+            catch (Exception ex)
+            {
+                // let's capture the exception
+                loadException = ExceptionDispatchInfo.Capture(ex);
+            }
+
+            // If the type cannot be found, and the name doesn't contain a version,
+            // we try to find the type in the current domain/alc using any assembly that has the same name.
+            if (type is null && !partialName.Contains("Version="))
+            {
+                var typePair = partialName.Split([','], StringSplitOptions.RemoveEmptyEntries);
+                if (typePair.Length != 2)
+                {
+                    if (throwOnError)
+                    {
+                        DuckTypeException.Throw($"Invalid type name: {partialName}");
+                    }
+
+                    return null;
+                }
+
+                var typeValue = typePair[0].Trim();
+                var assemblyValue = typePair[1].Trim();
+
+                try
+                {
+                    foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+                    {
+                        if (assembly.GetName().Name != assemblyValue)
+                        {
+                            continue;
+                        }
+
+                        type = assembly.GetType(typeValue, throwOnError: false);
+                        if (type is not null)
+                        {
+                            break;
+                        }
+                    }
+                }
+                catch
+                {
+                    // If we cannot get the assemblies, we just ignore them.
+                    // we are not interested in throwing an exception here.
+                    // we will just continue with an empty array.
+                    // And will throw later the original Type.GetType exception.
```

Comment:
well... _if_ `throwOnError == true`, otherwise we just return `null` as before, but after doing a lot more work 😬 I wonder if we should add an error log/throw here (or in the caller) because it _should_ never happen, and if it is, we're being a lot more expensive than before...

Reply:
changed in [5bcd7ae](https://github.com/DataDog/dd-trace-dotnet/pull/6284/commits/5bcd7aef97145bdad1950e684d0738196e282ef4) now we throw